ext/ftp/ftp.h

```diff
@@ -39,6 +39,11 @@
 /* XXX this should be configurable at runtime XXX */
 #define	FTP_BUFSIZE	4096
 
+#define FTP_MLSD_LINE_LENGTH_MAX 4096
+#define FTP_MLSD_CHUNK_LENGTH_MAX 256
```

Comment:
If the RFC doesn't specify limits, I think it would be preferable to not impose anything on our side. After all the limits are only checked after we already received the entire response, so they would not protect against malicious, overly large responses in any case.

You are currently making use of the chunk length limit for a fixed size buffer -- the need for this limit could be avoided by not copying into a buffer and instead only keeping a pointer to the start (you already store the length).